### PR TITLE
CHE-2518: add ability to add/remove stack's tags

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?

Add ability to add/remove stack's tags 
### What issues does this PR fix or reference?
#2518

tags

Signed-off-by: Oleksii Orel oorel@codenvy.com

### CODE REVIEW

File: dashboard/src/app/stacks/stack-details/stack.controller.ts [hunk 1]
```diff
@@ -16,12 +16,31 @@
...
+  stackContent: string;
+  invalidStack: string;
+
+  $log;
+  stack;
+  $filter;
+  cheStack;
+  $timeout;
+  $location;
+  $mdDialog;
```
Comment (benoitf) [inline dashboard/src/app/stacks/stack-details/stack.controller.ts]: for information, you may use https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/angular-material/angular-material.d.ts#L114 type
as we have angular-material definitions there https://github.com/eclipse/che/blob/master/dashboard/tsd.json#L26

the same for Angular object, you can use TypeScript definition (which are not any)

File: dashboard/src/app/stacks/stack-details/stack.controller.ts [hunk 2]
```diff
@@ -103,9 +122,35 @@ export class StackController {
   }
 
   /**
+   * Handle stack's tag adding.
+   *
+   * @param tag stack's tag
+   * @returns {*}
+   */
+  handleTagAdding(tag) {
```
Comment (benoitf) [inline dashboard/src/app/stacks/stack-details/stack.controller.ts]: you may add type of argument
like tag : any (or string I've not checked) and as there is a return value, you may add it at the end of the method

`handleTagAdding(tag : string) : string`

**Review by benoitf — APPROVED** [review]

File: dashboard/src/app/stacks/stack-details/stack.html [hunk 3]
```diff
@@ -47,6 +47,28 @@
     </div>
   </che-label-container>
 
+  <!-- stack's tags -->
+  <che-label-container class="stack-tags-label"
+                       che-label-name="Tags"
+                       che-label-description="Tags define templates available for workspace">
```
Comment (ashumilova) [inline dashboard/src/app/stacks/stack-details/stack.html]: "Tags define the templates ..." on the mockup

File: dashboard/src/app/stacks/stack-details/stack.html [hunk 4]
```diff
@@ -47,6 +47,28 @@
...
+                md-on-remove="stackController.updateEditorContent()"
+                secondary-placeholder="Enter stack's tags">
+        <md-chip-template>
+          <div class="stack-library-text">{{$chip | uppercase}}</div>
+          <div class="stack-library-btn">
+            <i class="fa fa-close"></i>
+          </div>
+        </md-chip-template>
+      </md-chips>
+      <div class="stack-tags-reset" data-ng-click="stackController.stack.tags.length = 0">Reset</div>
```
Comment (ashumilova) [inline dashboard/src/app/stacks/stack-details/stack.html]: Make Reset button not 100% width of the row - now click below the input causes clearing the tags.

File: dashboard/src/app/stacks/stack-details/stack.html (hunk 3 above)
Comment (olexii4) [reply dashboard/src/app/stacks/stack-details/stack.html]: OK

File: dashboard/src/app/stacks/stack-details/stack.controller.ts (hunk 2 above)
Comment (olexii4) [reply dashboard/src/app/stacks/stack-details/stack.controller.ts]: done

File: dashboard/src/app/stacks/stack-details/stack.controller.ts (hunk 1 above)
Comment (olexii4) [reply dashboard/src/app/stacks/stack-details/stack.controller.ts]: done

**Review by ashumilova — APPROVED** [review]

**Review by akurinnoy — APPROVED** [review]

**Pull request merged**